### ISSUE TITLE
Having Trouble installing certain dependencies

### ISSUE THREAD

**Issue by mhuang02** [sachindhital/Teaching_Assistant#1]
Hi there, I came across your project and I was interested in messing around with it but I am having some difficulties with running the software. For step 3 when I run the command, I am running into this error regarding langchain that I've attached below. I've been messing around with trying to downgrade certain dependencies so that it is compatible with conflict libraries but it just seems to create more compatibility issues with other dependencies. Here is my output after creating a fresh vm and running the command. (I've upgraded my pip version as well but that doesn't seem to fix anything). Any help would be appreciated!

> `Using cached zipp-3.21.0-py3-none-any.whl (9.6 kB)
Installing collected packages: flatbuffers, protobuf, namex, numpy, h5py, ml-dtypes, absl-py, mdurl, markdown-it-py, typing-extensions, pygments, rich, optree, packaging, keras, six, grpcio, idna, charset-normalizer, certifi, urllib3, requests, libclang, wheel, astunparse, tensorflow-io-gcs-filesystem, wrapt, opt-einsum, tensorboard-data-server, zipp, importlib-metadata, markdown, MarkupSafe, werkzeug, tensorboard, gast, termcolor, google-pasta, tensorflow-intel, tensorflow, async-timeout, numexpr, tenacity, greenlet, SQLAlchemy, pydantic, marshmallow, mypy-extensions, typing-inspect, dataclasses-json, langsmith, propcache, multidict, yarl, frozenlist, aiosignal, attrs, aiohappyeyeballs, aiohttp, openapi-schema-pydantic, PyYAML, langchain, jsonpointer, jsonpatch, langchain-core, langchain-community, filelock, fsspec, colorama, tqdm, huggingface-hub, tokenizers, safetensors, regex, transformers, catalogue, spacy-loggers, marisa-trie, language-data, langcodes, wasabi, cymem, murmurhash, preshed, jinja2, srsly, smart-open, pathlib-abc, click, typer, pathy, spacy-legacy, confection, blis, thinc, spacy, mpmath, sympy, networkx, torch, pytz, python-dateutil, tzdata, pandas, scipy, joblib, threadpoolctl, scikit-learn, faiss-cpu
ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

> We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.
> 
> langchain-core 0.1.53 requires langsmith<0.2.0,>=0.1.0, but you'll have langsmith 0.0.92 which is incompatible.
> langchain-core 0.1.53 requires packaging<24.0,>=23.2, but you'll have packaging 24.2 which is incompatible.
> Successfully installed MarkupSafe-3.0.2 PyYAML-6.0.2 SQLAlchemy-2.0.40 absl-py-2.2.2 aiohappyeyeballs-2.6.1 aiohttp-3.11.16 aiosignal-1.3.2 astunparse-1.6.3 async-timeout-4.0.3 attrs-25.3.0 blis-0.7.11 catalogue-2.0.10 certifi-2025.1.31 charset-normalizer-3.4.1 click-8.1.8 colorama-0.4.6 confection-0.1.5 cymem-2.0.11 dataclasses-json-0.5.14 faiss-cpu-1.7.4 filelock-3.18.0 flatbuffers-25.2.10 frozenlist-1.5.0 fsspec-2025.3.2 gast-0.6.0 google-pasta-0.2.0 greenlet-3.1.1 grpcio-1.71.0 h5py-3.13.0 huggingface-hub-0.30.2 idna-3.10 importlib-metadata-8.6.1 jinja2-3.1.6 joblib-1.4.2 jsonpatch-1.33 jsonpointer-3.0.0 keras-3.9.2 langchain-0.0.255 langchain-community-0.0.3 langchain-core-0.1.53 langcodes-3.5.0 langsmith-0.0.92 language-data-1.3.0 libclang-18.1.1 marisa-trie-1.2.1 markdown-3.7 markdown-it-py-3.0.0 marshmallow-3.26.1 mdurl-0.1.2 ml-dtypes-0.4.1 mpmath-1.3.0 multidict-6.4.2 murmurhash-1.0.12 mypy-extensions-1.0.0 namex-0.0.8 networkx-3.2.1 numexpr-2.10.2 numpy-1.24.2 openapi-schema-pydantic-1.2.4 opt-einsum-3.4.0 optree-0.15.0 packaging-24.2 pandas-2.1.2 pathlib-abc-0.1.1 pathy-0.11.0 preshed-3.0.9 propcache-0.3.1 protobuf-4.25.6 pydantic-1.10.21 pygments-2.19.1 python-dateutil-2.9.0.post0 pytz-2025.2 regex-2024.11.6 requests-2.32.3 rich-14.0.0 safetensors-0.5.3 scikit-learn-1.2.2 scipy-1.10.0 six-1.17.0 smart-open-6.4.0 spacy-3.6.1 spacy-legacy-3.0.12 spacy-loggers-1.0.5 srsly-2.5.1 sympy-1.13.3 tenacity-8.5.0 tensorboard-2.17.1 tensorboard-data-server-0.7.2 tensorflow-2.17.0 tensorflow-intel-2.17.0 tensorflow-io-gcs-filesystem-0.31.0 termcolor-3.0.1 thinc-8.1.12 threadpoolctl-3.6.0 tokenizers-0.21.1 torch-2.1.0 tqdm-4.67.1 transformers-4.51.2 typer-0.9.4 typing-extensions-4.13.2 typing-inspect-0.9.0 tzdata-2025.2 urllib3-2.4.0 wasabi-1.1.3 werkzeug-3.1.3 wheel-0.45.1 wrapt-1.17.2 yarl-1.19.0 zipp-3.21.0
> WARNING: You are using pip version 20.2.3; however, version 25.0.1 is available.
> You should consider upgrading via the 'c:\users\matt\teaching_assistant\venv\scripts\python.exe -m pip install --upgrade pip' command.`
